modules/features/shared/src/main/java/au/com/shiftyjelly/pocketcasts/shared/NetworkConnectionWatcher.kt

```diff
@@ -0,0 +1,63 @@
+package au.com.shiftyjelly.pocketcasts.shared
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkRequest
+import androidx.annotation.VisibleForTesting
+import au.com.shiftyjelly.pocketcasts.repositories.di.ApplicationScope
+import au.com.shiftyjelly.pocketcasts.repositories.playback.PlaybackManager
+import dagger.hilt.android.qualifiers.ApplicationContext
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import au.com.shiftyjelly.pocketcasts.utils.Network as PCNetworkUtils
+
+class NetworkConnectionWatcher @Inject constructor(
+    @ApplicationScope private val applicationScope: CoroutineScope,
+    @ApplicationContext private val context: Context,
+    private val playbackManager: PlaybackManager,
+) {
+
+    // Default to true so we do not detect a change to being metered when the class is initialized
+    private var connectionIsMetered: Boolean = true
+
+    private val networkCallback = object : ConnectivityManager.NetworkCallback() {
+        override fun onCapabilitiesChanged(network: Network, networkCapabilities: NetworkCapabilities) {
+            applicationScope.launch {
+                this@NetworkConnectionWatcher.onCapabilitiesChanged(networkCapabilities)
+            }
+            super.onCapabilitiesChanged(network, networkCapabilities)
+        }
+    }
+
+    fun startWatching() {
```

Comment:
Should we use a cold flow for watching the network connectivity? wdyt?

Reply:
We could do that, but I'm not seeing a lot of benefit to that change at this point. Since this is in one of the higher-level modules, most other classes cannot directly depend on this class (and observe the flow) without us adding a fair bit of DI complexity.

Reply:
I've updated the class monitoring the network status to just expose a flow. I just used a state flow because that seemed more straightforward and the PlaybackManager will be collecting from this flow during the entire life of the application anyway.